### ISSUE TITLE
Load/save in  Microsoft.ML from 1.3.1

### ISSUE THREAD

**Issue by nfnpmc** [dotnet/machinelearning#4103]
### System information

- **OS version/distro**:
Microsoft Windows 10 Enterprise
10.0.17763 Build 17763
- **.NET Version (eg., dotnet --info)**: 
Microsoft Windows [Version 10.0.17763.379]
(c) 2018 Microsoft Corporation. All rights reserved.

>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.508
 Commit:    9ba8583e91

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.508\

Host (useful for support):
  Version: 2.1.12
  Commit:  ccea2e606d

.NET Core SDKs installed:
  1.0.0-preview2-003131 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.507 [C:\Program Files\dotnet\sdk]
  2.1.508 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

### Issue

- **What did you do?**
Saved trained model
 context.Model.Save(model, data.Schema, modelLocation);
in the same class
ITransformer model = context.Model.Load(modelLocation, out DataViewSchema schema);
works OK.

In a different class
model = context.Model.Load(modelLocation, out DataViewSchema schema);
- **What happened?**
Got exception 
Could not load zip, something missing at the end of zip file, apparently.
- **What did you expect?**
A loaded model.
### Source code / logs
Then after 3 days I manually changed the csproj file as NuGet could not change the versions!
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
    <LangVersion>7.1</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.ML" Version="1.3.1" />
    <PackageReference Include="Microsoft.ML.CpuMath" Version="1.3.1" />
    <PackageReference Include="Microsoft.ML.DataView" Version="1.3.1" />
    <PackageReference Include="Microsoft.ML.Dnn" Version="0.15.1" />
    <PackageReference Include="Microsoft.ML.ImageAnalytics" Version="1.3.1" />
    <PackageReference Include="Microsoft.ML.TensorFlow" Version="1.3.1" />
    <PackageReference Include="Microsoft.ML.TensorFlow.Redist" Version="0.14.0" />
    <PackageReference Include="OpenCvSharp4" Version="4.1.0.20190416" />
    <PackageReference Include="OpenCvSharp4.runtime.win" Version="4.1.0.20190416" />
  </ItemGroup>

Replace 1.3.1 by 1.1.0

Retest - all worked
Changed them back to try to get the exact exception again.
All worked!

In a separate program, Loading the model originally saved before I changed the csproj file in the training program
NodeDef mentions attr 'explicit_paddings' not in Op<name=Conv2D; signature=input:T, filter:T -> output:T; attr=T:type,allowed=[DT_HALF, DT_BFLOAT16, DT_FLOAT, DT_DOUBLE]; attr=strides:list(int); attr=use_cudnn_on_gpu:bool,default=true; attr=padding:string,allowed=["SAME", "VALID"]; attr=data_format:string,default="NHWC",allowed=["NHWC", "NCHW"]; attr=dilations:list(int),default=[1, 1, 1, 1]>; NodeDef: {{node conv2d0_pre_relu/conv}}. (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
Error during class instantiation
Exception has been thrown by the target of an invocation.

Loading the new model works fine.
So I could not reproduce the issue I have been tring to solve for ages.
Oh, be joyful!

Please paste or attach the code or logs or traces that would be helpful to diagnose the issue you are reporting.

**Comment by najeeb-kazmi** [comment]
@nfnpmc what are the contents of the model pipeline / transformer chain you saved? Can you share the training code? From the exception, it looks like there are some DNNs involved in the pipeline so a generic pipeline that I could write probably won't be enough to try to reproduce.

Also, what other ML.NET nugets (and their versions) are installed in your project, other than Microsoft.ML? E.g. Microsoft.ML.TensorFlow or Microsoft.ML.ImageAnalytics.

**Comment by nfnpmc** [comment]
Blimey, that was over 6 months ago. Anyway it looks like getting all the ducks in line (versions compatible) e.g. Exe netcoreapp2.1 7.1, Replace 1.3.1 by 1.1.0, it all worked. It looks like it was a problem with NuGet not fetching the correct version. I never tried to break it again for you, sorry.
You can close this off now.
